File: app/src/main/java/com/rachelleignacio/listbucket/util/Keyboard.java
package com.rachelleignacio.listbucket.util;

import android.os.IBinder;
import android.view.inputmethod.InputMethodManager;

/**
 * Created by rachelleignacio on 8/9/17.
 */

public class Keyboard {
    private static final Keyboard ourInstance = new Keyboard();

    public static Keyboard getInstance() {
        return ourInstance;
    }

    private Keyboard() {
    }

    public void showKeyboard(InputMethodManager imm) {
        imm.toggleSoftInput(InputMethodManager.SHOW_FORCED, InputMethodManager.HIDE_IMPLICIT_ONLY);
    }

    public void hideKeyboard(InputMethodManager imm, IBinder windowToken) {
        imm.hideSoftInputFromWindow(windowToken, 0);
    }
}

File: app/src/main/kotlin/com/rachelleignacio/listbucket/util/Keyboard.kt
package com.rachelleignacio.listbucket.util

import android.content.Context
import android.util.Log
import android.view.View
import android.view.inputmethod.InputMethodManager

/**
 * Created by rachelleignacio on 8/9/17.
 */
object Keyboard {
    fun show(ctx: Context, view: View) {
        view.postDelayed( { getImmFromContext(ctx).showSoftInput(view, InputMethodManager.SHOW_IMPLICIT) }, 100)

    }

    fun hide(ctx: Context, view: View) {
        getImmFromContext(ctx).hideSoftInputFromInputMethod(view.windowToken, InputMethodManager.HIDE_IMPLICIT_ONLY)
    }

    private fun getImmFromContext(ctx: Context): InputMethodManager {
        return ctx.getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
    }
}